Update Create dependency to 6.0.0

rosellle opened the pull request:
This PR updates the Create (Forge) dependency to 6.0.0 as a class inside of Create was moved. Closes #2115.

Currently, the new Create Maven is not mirrored at https://maven.squiddev.cc/mirror so there is a hack for this in the meantime inside of buildSrc. Ideally the new files get mirrored as well.

This also required a splitting of CreateIntegration across the fabric and forge subprojects, as Create Fabric has not yet updated (and likely wont for some time).

rosellle committed:
chore: split CreateIntegration by forge/fabric for 6.0.0 update.

The 6.0.0 changes moved BlockMovementChecks to `com.simibubi.create.api.contraption`, but Create Fabric has not updated yet, thus the split of CreateIntegration across projects.

rosellle committed:
hack: temporary addition of https://maven.createmod.net to buildSrc as it is not yet mirrored at https://maven.squiddev.cc/mirror

user7564 commented:
I can confirm after compiling with these same changes on the 1.21.1 version that the changes work, thank you for your quick fix

user7564 commented:
I Will make a new pull request with the 1.21.1 changes and credit you and this PR, thank you

SquidDev committed:
Switch back to maven.squiddev.cc mirror

Exclude transitive deps from Create — legacy MDG strips these on 1.20,
but they are present on 1.21.

SquidDev commented:
Thank you!
